hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestExponentialRetryPolicy.java

```diff
@@ -285,6 +293,67 @@ public void testAbfsConfigConstructor() throws Exception {
     Assert.assertEquals("Delta backoff interval was not set as expected.", expectedDeltaBackoff, policy.getDeltaBackoff());
   }
 
+//  public void testClientBackoffOnlyNewRequest() throws IOException {
+@Test
+public void testClientBackoffOnlyNewWriteRequest() throws IOException, InterruptedException {
+  AzureBlobFileSystem fs = getFileSystem();
+  AbfsClient client = fs.getAbfsStore().getClient();
+  AbfsConfiguration configuration = client.getAbfsConfiguration();
+  Assume.assumeTrue(configuration.isAutoThrottlingEnabled());
+  AbfsCounters counters = client.getAbfsCounters();
+
+  URL dummyUrl = client.createRequestUrl("/", "");
+  String dummyMethod = AbfsHttpConstants.HTTP_METHOD_PUT;
+
+  AbfsRestOperationType testOperationType = AbfsRestOperationType.Append;
+
+  AbfsRestOperation restOp = new AbfsRestOperation(testOperationType, client, dummyMethod, dummyUrl, new ArrayList<>());
+
+  Long writeThrottleStatBefore = counters.toMap().get(AbfsStatistic.WRITE_THROTTLES.getStatName());
+  Thread.sleep(10000);
+  boolean appliedBackoff = restOp.applyThrottlingBackoff(0, testOperationType, counters);
```

Comment:
lets not just test this method.

lets test the whole executeRequest method. Have httpOperation mocked. And change the behaviour of httpOperation.processResponse.. Fail it for 2-3 times. and assert.